lib/pbench/agent/tool_data_sink.py

```diff
@@ -883,6 +877,12 @@ def __enter__(self):
                 "'tm_log_capture' thread started, processing Tool Meister logs ..."
             )
 
+        # Set up signal responder for state control
+        self.sig_resp = state_signals.SignalResponder(
+            redis_host=self.redis_host, redis_port=self.redis_port
+        )
```

Comment:
This is what I though was the desired functionality, though it may be worth discussing if there is a potential use-case for having two separate redis servers, maybe one only accessible to the agent components and accessible to the benchmark runners?

Reply:
In my mental model (which, albeit, doesn't seem to match reality, necessarily 😆), the State Signals facility is a black box which is independent of how Pbench uses Redis.  (That is, the fact that State Signals uses Redis is an implementation detail which would be beyond the scope of Pbench, except that we have to give it a host and port.)

So, this code which has State Signals using the same Redis server as the rest of the TM is OK but is, in some sense, a violation of encapsulation.  Still, if State Signals isn't going to create its own Redis server and is going to rely on one provided by Pbench, then the above is a perfectly reasonable way to go.